lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -783,8 +786,9 @@ namespace Js
                     currentChar = replaceStr[substitutionOffset + 2];
                     if (currentChar >= _u('0') && currentChar <= _u('9'))
                     {
-                        uint16 tempCaptureIndex = (10 * captureIndex) + (uint16)(currentChar - _u('0'));
-                        Assert(tempCaptureIndex < 100); // numeric value of 2-digit positive decimal number
+                        // Should also be unsigned; see captureIndex above.
+                        int tempCaptureIndex = (10 * captureIndex) + (int)(currentChar - _u('0'));
+                        Assert(0 <= tempCaptureIndex && tempCaptureIndex < 100); // numeric value of 2-digit positive decimal number
```

Comment:
nit: The previous assert is 0<= x <= 9 and this is 0 <= x < 100; any chance of making both <=<= or <=<?

Reply:
Added note to follow-up task -- the hope with this change is to get it landed asap to reduce the chances of the broken code making it out of DEV3.